### PR TITLE
TRI-33773 Switch to using RSE React library

### PR DESCRIPTION
**Pull request by mike-puzon-tri**

### CODE REVIEW

File: web/frontend/src/App.tsx [hunk 1]
```diff
@@ -1,46 +1,15 @@
...
-import Footer from '@/components/Footer';
-import FeedbackButton from '@/components/FeedbackButton';
-import './index.css';
-import './fonts.css';
+import { TRIApp } from '@toyota-research-institute/rse-react-library';
+import '@toyota-research-institute/rse-react-library/style.css';
+import config from './tri.app.config';
+import Prelogin from '@/pages/pre-login/pre-login';
 
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
```
Comment (graynorton-tri) [inline web/frontend/src/App.tsx:6]: `TRIApp` actually handles instantiating a `queryClient`, so I think you can skip this import and the associated logic in this file.

File: web/frontend/src/pages/About/About.tsx [hunk 2]
```diff
@@ -1,135 +1,109 @@
-import React, { useEffect } from 'react';
-import { Link } from 'react-router-dom';
+import { Link, ScrollRestoration } from 'react-router-dom';
```
Comment (graynorton-tri) [inline web/frontend/src/pages/About/About.tsx:1]: Still want to figure out why it isn't working to include this somewhere in library code, but we can save that for a future PR.

File: web/frontend/src/pages/About/Highlighter.tsx
Comment (graynorton-tri) [inline web/frontend/src/pages/About/Highlighter.tsx:1]: Similar to `<ScrollRestoration>`, want to figure out why you needed to pull this into the app code, but can wait for a future PR.

File: web/frontend/src/index.tsx [hunk 3]
```diff
@@ -1,13 +1,27 @@
+import { StrictMode } from 'react';
 import { createRoot } from 'react-dom/client';
+import { Route, RouterProvider, createBrowserRouter, createRoutesFromElements } from 'react-router-dom';
+import { ThemeProvider } from '@material-tailwind/react';
```
Comment (graynorton-tri) [inline web/frontend/src/index.tsx:4]: This is really more of a note to myself: I didn't include `<ThemeProvider>` in either the library code or in the Convex Materials Design app code, and things seemed to be working fine without it. I'd like to understand better what it's doing and likely move it into the library if it's in fact needed. But again, can wait for a future PR.

**Review by graynorton-tri — APPROVED** [review]
Made a few comments to flag things I'd like to loop back and address in future PRs, but looks good!

File: web/frontend/src/App.tsx (hunk 1 above)
Comment (mike-puzon-tri) [reply web/frontend/src/App.tsx:6]: I had removed this in the react library cause I wasn't sure if all projects will use `use-query`. We can add it back later if it seems all projects use it anyways.

File: web/frontend/src/pages/About/Highlighter.tsx
Comment (mike-puzon-tri) [reply web/frontend/src/pages/About/Highlighter.tsx:1]: I might bring this back with the changes I made.

**Pull request merged**